lib/tasks/one_time/2019_Q2/set_when_approved.rake

```diff
@@ -0,0 +1,39 @@
+# Set the value for shf_application.when_approved
+#
+# This value needs to be initialized for all ShfApplications that are already
+# in the database and approved.
+#
+# This should have been done in the migration that added the 'when_approved' column to ShfApplication
+
+namespace :shf do
+  namespace :one_time do
+
+    desc "Set the ShfApplication when_approved data"
+    task set_when_approved_data: :environment do
+
+      log_file = 'log/set_when_approved.log'
+
+      ActivityLogger.open(log_file, 'OneTimeRakeTask', 'set_when_approved_data') do |log|
+
+        approved_apps = User.includes(:shf_application).where.not(shf_application: nil).where(shf_applications: {state: 'accepted'}).distinct
+        log.record('info', "Found #{approved_apps.size} approved applications.")
+
+        members_shf_nil_approved_date = approved_apps.select{|m| m.shf_application.when_approved.nil?}
+        log.record('info', "Found #{members_shf_nil_approved_date.size} of those with no date for when_approved.")
+
+        members_shf_nil_approved_date.each do | member |
+          shf_app = member.shf_application
+          last_updated = shf_app.updated_at
+
+          if shf_app.update(when_approved: last_updated)
```

Comment:
We'll have to run this _after_ my current bug fix is deployed (app update will fail if file-delivery method is `nil` in current logic). #624

Reply:
Aha -- When I click on "files" to see the files changed by my PR, I then searched for `.rake` and found 0 of them.  But when I search for just `rake` with no period, I see the this comes up!

I'll have to see why this  changed in my PR.  It shouldn't be.

Reply:
This rake file was moved into a different directory (`2019_Q1`), but because I started this branch earlier, I still have the file where it was previously (in the `2019_Q2` folder.  I've updated my branch.